src/vcf/validator.cpp

```diff
@@ -222,17 +249,18 @@ namespace ebi
     }
 
     bool validate(const std::vector<char> &firstLine,
-                  std::istream &input,
+                  boost::iostreams::filtering_istream &input,
```

Comment:
I'd say filtering `filtering_istream` extends `std::istream`, so we can keep asking for the interface here.